src/openforms/logging/logevent.py

```diff
@@ -439,29 +441,110 @@ def submission_export_list(form: "Form", user: "User"):
+            LogicActionTypes.value: "component",
+            LogicActionTypes.property: "component",
+            LogicActionTypes.variable: "variable",
+        }
+        for action in rule.actions:
+            action_details = action["action"]
+            component_key = action.get(component_key_lookup.get(action_details["type"]))
+            component_meta = components_map.get(component_key)
+            component = component_meta.component if component_meta else None
+
+            # figure out the best possible label
+            # 1. fall back to component label if there is a label, else empty string
+            # 2. if there is a variable, use the name if it's set, else fall back to
+            # component label
+            label = component.get("label", "") if component else ""
+            if component_key in all_variables:
+                label = all_variables[component_key].name or label
+
+            action_log_data = {
+                "key": component_key,
+                "type_display": LogicActionTypes.get_choice(
+                    action_details["type"]
+                ).label,
+                "label": label,
+                "step_name": component_meta.form_step.form_definition.name
+                if component_meta
+                else "",
+                "value": "",
+            }
+
+            # process the value
+            if action_details["type"] == LogicActionTypes.value:
+                action_log_data["value"] = action_details["value"]
+            elif action_details["type"] == LogicActionTypes.property:
+                action_log_data.update(
+                    {
+                        "value": action_details["property"]["value"],
+                        "state": action_details["state"],
+                    }
+                )
+            elif action_details["type"] == LogicActionTypes.variable:
+                action_logic_introspection = introspect_json_logic(
+                    action_details["value"], components_map, resulting_data
+                )
+                action_log_data["value"] = action_logic_introspection.as_string()
+            targeted_components.append(action_log_data)
 
         evaluated_rule_data = {
-            "source_components": rule.json_logic_trigger,
-            "targeted_components": rule.actions,
+            "raw_logic_expression": rule_introspection.expression,
+            "readable_rule": rule_introspection.as_string(),
+            "targeted_components": targeted_components,
             "trigger": evaluated_rule["trigger"],
         }
 
-        log_evaluated_rules.append(evaluated_rule_data)
+        evaluated_rules_list.append(evaluated_rule_data)
+
+    # de-duplication of input data
+    de_duplicated_input_data = [
+        dict(data_tuple) for data_tuple in {tuple(data.items()) for data in input_data}
+    ]
+
+    # Mapping input data with its key
+    input_data = {data["key"]: data for data in de_duplicated_input_data}
```

Comment:
took me a while to get what is going on here :P

I think a more readable version is:

```py
    deduplicated_input_data = {node["key"]: node for node in input_data}
```

You rely on the key-uniqueness for de-duplication. If a node occurs multiple times, it will simply overwrite an earlier assignment and that's fine, because they are identical anyway.

Then you can pass `deduplicated_input_data` to the `_create_log` call directly as well.